src/Workspaces/Core/Portable/CodeGeneration/AbstractCodeGenerationService_FindDeclaration.cs

```diff
@@ -164,16 +172,20 @@ class NestedType
                 }
             }
 
-            // Generate into any decl we can find.
+            // If there is a declaration in a non auto-generated file, prefer it.
+            Func<Document,bool> isGeneratedCode = 
+                solution.Workspace.Services.GetService<IGeneratedCodeRecognitionService>().IsGeneratedCode;
```

Comment:
Correct, but WinForms does generate code other than the contents of InitializeComponent. It's just that it intentionally stomps on changes to InitializeComponent. I'm being pedantic Kean! :smile: